### DISCUSSION TITLE
No connection to TCP-Serial Bridge to ZIGBEE2MQTT and HZA

### DISCUSSION THREAD

**Discussion by prior-philip** [arendst/Tasmota#21625] [Support request] [answered]
### PROBLEM DESCRIPTION
Neither zigbee2mqtt nor HZA is able to successfully connect to zigbee bridge **pro**.
Console  shows:

> 13:13:12.053 TCP: Got connection from 192.168.1.13

but zigbee2mqtt throws

> [2024-06-13 14:13:12] warning:  zh:ezsp: Deprecated driver 'ezsp' currently in use, 'ember' will become the officially supported EmberZNet driver in next release. If using Zigbee2MQTT see https://github.com/Koenkk/zigbee2mqtt/discussions/21462
[2024-06-13 14:13:22] error:    zh:ezsp:uart: --> Error: Error: {"sequence":-1} after 10000ms
Error: Failure to connect
    at SerialDriver.resetForReconnect (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:347:19)
    at SerialDriver.emit (node:events:519:28)
    at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:348:22
    at Queue.execute (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/queue.ts:35:20)
    at Socket.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:154:17)

I followed this process with no success.
https://github.com/arendst/Tasmota/discussions/17998#discussioncomment-5027394

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____Sonoff Zigbee Bridge Pro
- [x] Tasmota binary firmware version number used: _____14.1.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: _____ Tasmota Webinstaller
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
13:24:06.871 RSL: RESULT = {"NAME":"TCP ZBBridge Pro","GPIO":[0,0,576,0,480,0,0,0,0,1,1,224,0,0,0,5472,0,320,5793,5504,0,640,608,32,0,0,0,0,0,1,0,0,0,0,0,0],"FLAG":0,"BASE":1}
13:24:07.101 RSL: RESULT = {"Module":{"0":"TCP ZBBridge Pro"}}
13:24:07.359 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"576":"LedLink_i"},"GPIO3":{"0":"None"},"GPIO4":{"480":"Buzzer"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"224":"Relay1"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"5472":"TCP Tx"},"GPIO20":{"0":"None"},"GPIO21":{"320":"Led_i1"},"GPIO22":{"5793":"Zigbee Rst2"},"GPIO23":{"5504":"TCP Rx"},"GPIO24":{"0":"None"},"GPIO25":{"640":"I2C SDA1"},"GPIO26":{"608":"I2C SCL1"},"GPIO27":{"32":"Button1"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
13:25:01.083 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":75,"Free":436,"Rules":"on system#boot do TCPStart 8888 endon on system#boot do power1 toggle endon"}}
13:25:01.313 RSL: RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:25:01.562 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
13:25:23.719 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_4A4A58","ButtonTopic":"0","Power":"0","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
13:25:23.727 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"https://ota.tasmota.com/tasmota32/tasmota32-zbbrdgpro.bin","RestartReason":"Software reset CPU","Uptime":"0T00:19:54","StartupUTC":"2024-06-13T12:05:29","Sleep":50,"CfgHolder":4617,"BootCount":5,"BCResetTime":"2024-06-13T12:33:45","SaveCount":19}}
13:25:23.734 RSL: STATUS2 = {"StatusFWR":{"Version":"14.1.0(release-zbbrdgpro)","BuildDateTime":"2024-06-03T11:55:40","Core":"3_0_0","SDK":"5.1.4.240529","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 v3.1","CR":"416/699"}}
13:25:23.739 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["BONNA_IoT",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
13:25:23.751 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":1571,"Free":284,"Heap":148,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640C8","FlashFrequency":40,"FlashMode":"DIO","Features":["0809","8F8A87CF","00548001","000000CF","010013C0","C0000981","40024080","00200000","5400082C","00000080","00000004"],"Drivers":"1,2,!3,!4,!5,!8,9,10,12,!16,!20,!21,!23,24,!27,29,!35,38,41,50,52,!59,62,!68,82,!121","Sensors":"1,2,3,5,6,127","I2CDriver":"7"}}
13:25:23.762 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-4A4A58-2648","IPAddress":"192.168.1.199","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"2003:e2:bf47:7900:de39:6fff:fe18:506","Mac":"E4:65:B8:4A:4A:58","IP6Global":"fdd3:c9a5:591a:0:e665:b8ff:fe4a:4a58","IP6Local":"fe80::e665:b8ff:fe4a:4a58%st1","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"192.168.1.1","DNSServer2":"2003:e2:bf47:7900:de39:6fff:fe18:506","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0}}
13:25:23.770 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_4A4A58","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
13:25:23.776 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-06-13T12:25:23Z","Local":"2024-06-13T13:25:23","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"04:46","Sunset":"20:54"}}
13:25:23.784 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-06-13T13:25:23"}}
13:25:23.790 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-06-13T13:25:23","Uptime":"0T00:19:54","UptimeSec":1194,"Heap":151,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":46},"POWER":"OFF","Wifi":{"AP":1,"SSId":"BONNA_IoT","BSSId":"9E:9D:7E:74:AA:C9","Channel":1,"Mode":"HT20","RSSI":82,"Signal":-59,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
13:26:04.779 WIF: Checking connection...
13:26:17.005 TCP: Got connection from 192.168.1.13
13:26:17.009 TCP: to MCU/1: 1AC038BC7E
13:26:24.794 WIF: Checking connection...

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Start Serial to TCP Bridge on 8888 and invoke zigbee2mqtt on same port with adapter option "ezsp"

> serial:
  port: tcp://192.168.1.199:8888
  adapter: ezsp

### EXPECTED BEHAVIOUR
zigbee2mqtt (or HZA) to connect on 8888 and take over coordinator role

**Comment by Jason2866** [comment]
See https://github.com/arendst/Tasmota/issues/21555

**Reply by prior-philip** [reply]
Different device, hardware and firmware....
Im opening a topic for "Sonoff Zigbee Bridge **Pro**"

**Comment by s-hadinger** [comment]
From your logs the connection succeeds, but the MCU does not respond. Maybe a hardware or configuration problem. Are you sure your EZSP version is supported by Z2M or ZHA?

**Comment by prior-philip** [comment]
Well thats probably an open question. Not sure how to answer this.
coordinator fw 

> Zigbee: ZNP v2.7.1.20220219
is installed.

Newer version shows same behavior.

**Reply by s-hadinger** [reply]
Aha. ZNP is not EZSP. Pleased choose ZNP in Z2M or ZHA

**Comment by prior-philip** [comment]
ZNP seems not to be a valid option for zigbee2mqtt - ezsp and ember only
ZHA is supporting ZNP but doesnt connect either - even though the console is full of "from/to MCU..." messages.

> 6:03:01.960 TCP: to MCU/1: FE082133010100F00000000CE6
16:03:02.009 TCP: from MCU: FE0E6133000CFFFFFFFFFFFFFFFFFFFFFFFF50
16:03:02.059 TCP: to MCU/1: FE052132010100F100E7
16:03:02.108 TCP: from MCU: FE0461320C0000005B
16:03:02.160 TCP: to MCU/1: FE082133010100F10000000CE7
16:03:02.209 TCP: from MCU: FE0E6133000CFFFFFFFFFFFFFFFFFFFFFFFF50
16:03:02.259 TCP: to MCU/1: FE052132010100F200E4
16:03:02.308 TCP: from MCU: FE0461320C0000005B
16:03:02.360 TCP: to MCU/1: FE082133010100F20000000CE4
16:03:02.409 TCP: from MCU: FE0E6
16:03:02.459 TCP: to MCU/1: FE052132010100F300E5
16:03:02.508 TCP: from MCU: FE0461320C0000005B
16:03:02.560 TCP: to MCU/1: FE082133010100F30000000CE5
16:03:02.609 TCP: from MCU: FE0E6133000CFFFFFFFFFFFFFFFFFFFFFFFF50
16:03:02.659 TCP: to MCU/1: FE052132010100F400E2
16:03:02.708 TCP: from MCU: FE0461320C0000005B
16:03:02.760 TCP: to MCU/1: FE082133010100F40000000CE2
16:03:02.809 TCP: from MCU: FE0E6133000CFFFFFFFFFFFFFFFFFFFFFFFF50
16:03:02.859 TCP: to MCU/1: FE052132010100F500E3
16:03:02.908 TCP: from MCU: FE0461320C0000005B
16:03:16.607 TCP: to MCU/1: FE
16:03:21.116 WIF: Checking connection...

Unfortunately ZHA is not very informative as it just reports "ERROR" with no further details.

**Reply by s-hadinger** [reply]
Well they do "communicate" with the ZNP protocol. Now the problem is within ZHA and we can’t help more

**Reply by s-hadinger** [reply, marked as answer]
ZNP is supported by Zigbee2Mqtt , it is called zstack

**Reply by prior-philip** [reply]
Yeah, getting it hardly to work, its connecting every 10th time.

Thanks for your help. I see issues with pairing now - but for that I will create another case.

**Reply by prior-philip** [reply]
Oh goodness.....
Now I got the root cause for the crazy bahviour of this bridge with connecting and pairing.
The piggyback board has a soldering issue. Slightly pressing it, makes it all work. Need soldering probably.

**Reply by Elmbob** [reply]
> ZNP is supported by Zigbee2Mqtt , it is called zstack

Does that mean that in the configuration I am using `adapter: zstack` instead of `adapter: ezsp`? 

Edit: I just found the answer under https://github.com/arendst/Tasmota/discussions/17998
